http/basic/src/main/java/io/cloudevents/http/HttpMessageReader.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2018-Present The CloudEvents Authors
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.cloudevents.http;
+
+import io.cloudevents.SpecVersion;
+import io.cloudevents.core.message.impl.BaseGenericBinaryMessageReaderImpl;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+import static io.cloudevents.http.CloudEventsHeaders.CE_PREFIX;
+
+public class HttpMessageReader extends BaseGenericBinaryMessageReaderImpl<String, String> {
+
+    private final Iterable<Map.Entry<String,String>> headers;
+
+    protected HttpMessageReader(SpecVersion version, Iterable<Map.Entry<String,String>> headers, byte[] body) {
+        super(version, body);
+        this.headers = headers;
+    }
+
+    @Override
+    protected boolean isContentTypeHeader(String key) {
+        return key.equalsIgnoreCase("Content-Type");
+    }
+
+    @Override
+    protected boolean isCloudEventsHeader(String key) {
+        return key.length() > 3 && key.substring(0, CE_PREFIX.length()).toLowerCase().startsWith(CE_PREFIX);
+    }
+
+    @Override
+    protected String toCloudEventsKey(String key) {
+        return key.substring(CE_PREFIX.length()).toLowerCase();
+    }
+
+    @Override
+    protected void forEachHeader(BiConsumer<String, String> fn) {
```

Comment:
If the type of `headers` is `Map<String, String>` then this method is just `headers.forEach(fn)`.